flake.nix

```
@@ -0,0 +1,65 @@
+{
+  description = "A flake for the syn data fidelity library";
+
+  inputs = {
+    linux-nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
+    darwin-nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+  };
+
+  outputs = { self, linux-nixpkgs, darwin-nixpkgs, flake-utils, ... }: let
+    pkgs = {
+      x86_64-linux = linux-nixpkgs.legacyPackages.x86_64-linux;
+      aarch64-darwin = darwin-nixpkgs.legacyPackages.aarch64-darwin;
+    };
+    mkLPMfidelity = pkgs: (pkgs.python310Packages.buildPythonPackage {
+          pname = "lpm_fidelity";
+          version = "0.0.1";
+          src = self;
+          nativeBuildInputs = with pkgs.python310Packages; [
+            setuptools
+          ];
+          nativeCheckInputs = with pkgs.python310Packages; [
+            pytest
+            numpy
+            scipy
+            polars
+          ];
+          checkPhase = "pytest tests/ -vvv";
+          propagatedBuildInputs = (with pkgs.python310Packages; [
+            numpy
```

```suggestion
            pytest
            numpy
```

The directions in the README, to run the tests with `nix develop -c  pytest tests/ -vvv`, fails for me unless pytest is added to `propagatedBuildInputs`

---

ah right. I think I want `pytest in the output of the devshell, then.

---

done.